### ISSUE TITLE
question about low ev identification performance vs high final classification performance 

### ISSUE THREAD

**Issue by zhaoxy92** [jayded/evidence-inference#18]
Hi, 

Thanks for the work on Evidence Inference 2.0! After reading your paper, I have this question that I couldn't understand. 

The pipeline steps are:
 (1) evidence identification, which is a binary classification task like f(ICO, sentence) -> 1/0. Negative training data are sampled.
 (2) final classification like g(ICO, predicted ev) -> [increase, no difference, decrease]. You used gold ev to train and predicted ev sentence that has the highest probability score (from softmax) to do inference.

What I couldn't understand is that:  
The Table 2 and Table 3 in the paper report that your best identification accuracy is only around 0.67, which means that the selected evidence is like 67% to be actual evidence. In this case, during the final inference, there will be quite a lot of selected evidence that is actually not bearing evidence. However, the final performance reported in Table 1 is around 0.78 for precision, recall, and F1. Could you please explain what may cause this? Or, if I interpreted the pipeline wrong, could you please let me know? 

Thank you!

**Comment by jayded** [comment]
Hi @zhaoxy92 ,

Thanks for the question! I think your overall understanding is correct, and you've stumbled across a subtle point:
1. The data is _not_ exhaustively annotated. Most ICOs have only one or two supporting sentences identified, whereas they typically have four or so in the entire document. I believe we have some evidence for this; Ben Nye collected fully annotated abstracts for a small portion of documents in [this publication](https://www.semanticscholar.org/paper/Understanding-Clinical-Trial-Reports%3A-Extracting-Nye-DeYoung/673798ee254cc9e04df2bc3570b017560b8e4571). This means that _other_ evidence sentences for an ICO may be returned, but not be reflected in our accuracy statistics. I've hand-inspected a fair amount of the evidence sentences returned by the identifier; it's hard to find a mistake.
2. I believe some outcomes in the trials are correlated. I haven't measured this, but that would also explain this issue.

It's hard to know which of the two is more important in explaining the phenoma you mention.

Please let me know if you have additional questions,

Jay

**Comment by zhaoxy92** [comment]
Thank you for the quick response!
Another question I have about the dataset is:

The paper mentioned that, even with the Evidence Inference 1.0, the pipeline performance is similar with only a slight drop (BR Pipeline 1.0). Since you also authored the ERASER benchmark paper, I'm wondering if the evidence inference dataset in the ERASER paper is the same as the Evidence Inference 1.0 data that you used in this paper. I'm asking this because I've been using the dataset in the ERASER paper since the data there has been preprocessed with a nice format (tokenized sentence-wise).

Thank you!

**Comment by jayded** [comment]
The Evidence Inference in ERASER is the same as the version 1.0, except:
1. A small subset of the data has exhaustive annotations collected, see Appendix B of the [ERASER paper](https://arxiv.org/pdf/1911.03429.pdf). I believe these have a special marker in the `query_type` field. I believe these are available in the current version of the evidence inference repo as well as in ERASER.
2. minor differences from conversion that I don't think are important. Specifically: sentence boundaries were reverse engineered from original text offsets and sentence breaking over biomedical text is...not great; some whitespace and text normalization. All annotations passed a sanity check I wrote verifying the original text closely matched the result for ERASER.

I have considered updating the evidence inference annotations to be available in the ERASER format, but I don't have time to look into that for about a month.

**Comment by zhaoxy92** [comment]
@jayded Thanks for the clarification. It's just that I've been using the ERASER version of the dataset and can't reproduce the performance. The oracle performance is similar enough (about 0.8), but the inference with the top-ranked evidence sentence is pretty far (only ~0.6).  Although I'm using my own implementation, but I don't think it should matter because the pipeline is pretty straightforward as following:
    evidence identifier: RoBERTa(sentence, query) -> 1 or 0
    evidence classification: RoBERTa(evidence, query) -> 0, 1, or 2.

I've tried bioBERT and pubmedBERT, and both have similar results. 

Is there any particular preprocessing that should be done?  So far the only preprocessing I have is that, for the query, I use three special tokens to create the query:  [I] intervention [C] comparator [O] outcome.

**Comment by jayded** [comment]
I don't have much bandwidth to help you debug code remotely, but:
1. [in the Evidence Inference code prepares a single input instance here](https://github.com/jayded/evidence-inference/blob/master/evidence_inference/models/pipeline.py#L834) ;  [within the model the query and sentence are concatenated here](https://github.com/jayded/evidence-inference/blob/master/evidence_inference/models/bert_model.py#L103) ; the special tokens are likely important; it's unclear if the position tokens are important, but they are handled properly.
2. negatives are sampled uniquely *per epoch*, so the model actually sees many negative candidates, but few times each candidate. I imagine this is important but I never measured it.

Both the evidence inference code and the ERASER code should more or less "just reproduce" after setting up the appropriate venv and running the documented commands.

**Comment by zhaoxy92** [comment]
Ok thanks. I'll make these two modifications to see if they would help.

**Comment by jayded** [comment]
I'm going to close this issue as stale - please reopen if not.
